### PR TITLE
[SPARK-40267][DOC] Add description for ExecutorAllocationManager metrics

### PR DESCRIPTION
**Pull request by warrenzhu25**
### What changes were proposed in this pull request?
Add description for ExecutorAllocationManager metrics

### Why are the changes needed?
Some ExecutorAllocationManager metrics are hard to know what stands for just from metric name.

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
No code change.

### CODE REVIEW

**Comment by warrenzhu25** [comment]
@dongjoon-hyun Could you help take a look?

**Comment by warrenzhu25** [comment]
@Ngone51 Could you help take a look?

File: docs/monitoring.md [hunk 1]
```diff
@@ -1207,12 +1207,12 @@ This is the component with the largest amount of instrumented metrics
...
-  - executors.numberAllExecutors
-  - executors.numberTargetExecutors
-  - executors.numberMaxNeededExecutors
-  - executors.numberDecommissioningExecutors
+  - executors.numberExecutorsToAdd: Number of executors to add in the next round sync with cluster manager
+  - executors.numberExecutorsPendingToRemove: Number of executors to remove in the next round sync with cluster manager
+  - executors.numberAllExecutors: Number of all running executors including normal and decommissioning
+  - executors.numberTargetExecutors: Desired number of executors totally needed from cluster manager
+  - executors.numberMaxNeededExecutors: Maximum number of executors needed under the current load to satisfy all running and pending tasks, rounded up.
+  - executors.numberDecommissioningExecutors: Number of executors are decommissioning
```
Comment (dongjoon-hyun) [inline docs/monitoring.md:1215]: We don't have comments in this way in this doc, do we, @warrenzhu25 ? I'm wondering if this is a right place to have these comments.

File: docs/monitoring.md [hunk 2]
```diff
@@ -1207,12 +1207,12 @@ This is the component with the largest amount of instrumented metrics
...
     parameter `spark.dynamicAllocation.enabled` (default is false)
-  - executors.numberExecutorsToAdd  
-  - executors.numberExecutorsPendingToRemove
-  - executors.numberAllExecutors
-  - executors.numberTargetExecutors
-  - executors.numberMaxNeededExecutors
-  - executors.numberDecommissioningExecutors
+  - executors.numberExecutorsToAdd: Number of executors to add in the next round sync with cluster manager
+  - executors.numberExecutorsPendingToRemove: Number of executors to remove in the next round sync with cluster manager
+  - executors.numberAllExecutors: Number of all running executors including normal and decommissioning
```
Comment (dongjoon-hyun) [inline docs/monitoring.md:1212]: This looks misleading because `decommissioning` is just a status.

File: docs/monitoring.md (hunk 1 above)
Comment (warrenzhu25) [reply docs/monitoring.md:1215]: Yes, most metrics are self-explanation, but these metrics are not. Putting comments just after metrics name will be easier for users to understand. If you think there's better place, I'm happy to move it around.

**Pull request closed without merging**